gst_plugins/python/avro_video_demux.py

```diff
@@ -263,14 +265,21 @@ def handle_video_frame(self, message: Dict, buffer: Gst.Buffer) -> Gst.FlowRetur
                     self.max_parallel_streams
                     and len(self.sources) >= self.max_parallel_streams
                 ):
-                    self.logger.warning(
-                        'Reached maximum number of streams: %s. '
-                        'Skipping frame %s from source %s',
-                        self.max_parallel_streams,
-                        frame_pts,
-                        source_id,
+                    error = (
+                        f'Failed to add source {source_id!r}: reached maximum '
+                        f'number of streams: {self.max_parallel_streams}.'
                     )
-                    return Gst.FlowReturn.OK
+                    self.logger.error(error)
+                    frame = inspect.currentframe()
+                    propagate_gst_error(
+                        gst_element=self,
+                        frame=frame,
+                        file_path=__file__,
+                        domain=Gst.StreamError.quark(),
+                        code=Gst.StreamError.DEMUX,
+                        text=error,
+                    )
+                    return Gst.FlowReturn.ERROR
```

Comment:
Sometimes the module hangs when started simultaneously with `max_parallel_streams+1` sources.

```
2023-05-19 11:39:34,005 [ERROR] [savant.avro_video_demux] [Dummy-4] Failed to add source 'city-traffic-015ec1b103e3': reached maximum number of streams: 2.
2023-05-19 11:39:34,006 [ERROR] [savant.gstreamer.runner] [Thread-3] Received error "gst-stream-error-quark: Failed to add source 'city-traffic-015ec1b103e3': reached maximum number of streams: 2. (9)" from avro_video_demux+avrovideodemux0.
[NvMultiObjectTracker] De-initialized
```

Reply:
The module hangs when it tries to add elements or request pads after `shutdown()` was called. Need to check pipeline is running before adding elements or requesting pads.